filebeat/fileset/config.go

```diff
@@ -13,5 +20,24 @@ type ModuleConfig struct {
 type FilesetConfig struct {
 	Enabled    *bool                  `config:"enabled"`
 	Var        map[string]interface{} `config:"var"`
+	Input      map[string]interface{} `config:"input"`
 	Prospector map[string]interface{} `config:"prospector"`
 }
+
+// NewFilesetConfig creates a new FilesetConfig from a common.Config.
+func NewFilesetConfig(cfg *common.Config) (*FilesetConfig, error) {
+	var fcfg FilesetConfig
+	err := cfg.Unpack(&fcfg)
+	if err != nil {
+		return nil, fmt.Errorf("error unpacking configuration")
+	}
+
+	if len(fcfg.Prospector) > 0 {
```

Comment:
Tests ...

Reply:
I have added unit test for this path, which are a lot simplier than integration test.

Reply:
There is also test there https://github.com/elastic/beats/pull/6078/files/c2b149f7d67b082b20f8201db719ba1c01755683#diff-68d15c248f81d1c5e67c59c6dab45574R59

Reply:
The part which I think unit tests here can't cover is to check if it's logged correctly. +1 on the unit test.